src/js/plugins/tts/FestivalTTSEngine.js

```diff
@@ -72,41 +73,24 @@ export default class FestivalTTSEngine extends AbstractTTSEngine {
      * @return {PromiseLike}
      */
     playChunk(chunk) {
+        if (!chunk.sound.loaded) this.opts.onLoadingStart();
         return new Promise(res => chunk.sound.play({ onfinish: res }))
         .then(() => chunk.sound.destruct());
     }
 
     /**
      * @private
      * @param {PageChunk} pageChunk
-     * @return {PromiseLike<PageChunk & { sound: SMSound }>}
+     * @return {PageChunk & { sound: SMSound }}
      */
     fetchChunkSound(pageChunk) {
-        this.opts.onLoadingStart();
-        return new Promise(res => {
-            const resolve = sound => {
-                this.opts.onLoadingComplete();
-                res($.extend(pageChunk, { sound }))
-            };
-            soundManager.createSound({
-                url: this.getSoundUrl(pageChunk.text),
-
-                // Many soundManger2 callbacks are broken when using HTML5 audio.
-                // whileloading: broken on safari, worked in FF4, but broken on FireFox 5
-                // onload: fires on safari, but *after* the sound starts playing, and does not fire in FF or IE9
-                // onbufferchange: fires in FF5 using HTML5 audio, but not in safari using flash audio
-                // whileplaying: fires everywhere
-                onload: function() { resolve(this); },
-                //fires in FF and IE9
-                onready: function() { resolve(this); },
-                //fires in safari...
-                onbufferchange: function() {
-                    if (!this.isBuffering) {
-                        resolve(this);
-                    }
-                }
-            }).load();
+        pageChunk.sound = soundManager.createSound({
+            url: this.getSoundUrl(pageChunk.text),
+            // API recommended, but only fires once play started on safari
+            onload: () => this.opts.onLoadingComplete()
```

Comment:
SUGGESTION: For AirBNB style, add a comma.

Reply:
Done